### PR TITLE
Add docker-compose support

### PR DESCRIPTION
**Pull request by flepied**
Issue: https://github.com/morucci/monocle/issues/19

### CODE REVIEW

File: monocle/db/db.py [hunk 1]
```diff
@@ -19,20 +19,33 @@
...
-
 
 class ELmonocleDB():
 
-    def __init__(self, index='monocle'):
-        self.es = client.Elasticsearch('localhost:9200')
+    def __init__(self, elastic_conn='localhost:9200', index='monocle'):
+        while True:
+            try:
+                print('Connecting to ES server at %s' % elastic_conn)
```
Comment (morucci) [inline monocle/db/db.py]: You could use a logger.

File: monocle/envdefault.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+import argparse
```
Comment (morucci) [inline monocle/envdefault.py]: Once the generic crawler is merged, we should not need this env vars input. The configuration should be passed through a config yaml file. And called with: monocle crawler --config config.yaml

Comment (morucci) [reply monocle/envdefault.py]: But, this could be refactored after we merge both PRs.

File: monocle/webapp.py [hunk 3]
```diff
@@ -30,14 +31,15 @@
...
+CORS(app, resources={r"/api/0/query/*":
+                     {"origins": os.getenv('ALLOW_ORIGIN', '*')}})
 
 
 @app.route("/api/0/query/<name>", methods=['GET'])
 def query(name):
     repository_fullname = request.args.get('repository')
     params = utils.set_params(request.args)
-    db = ELmonocleDB()
+    db = ELmonocleDB(os.getenv('ELASTIC_CONN', 'localhost:9200'))
```
Comment (morucci) [inline monocle/webapp.py]: As the api will be nested in the monocle cli, we could remove the env vars input as well.

Comment (morucci) [reply monocle/webapp.py]: But again can be fixed wth a followup pr.

File: README.md [hunk 4]
```diff
@@ -14,60 +14,37 @@ cd monocle
 mkdir data
 ```
```
Comment (morucci) [inline README.md]: chmod o+w data

w/o it elasticsearch does not start. Forbidden to write in data.

File: monocle/envdefault.py (hunk 2 above)
Comment (flepied) [reply monocle/envdefault.py]: sure it depends which one we merge first...

File: README.md (hunk 4 above)
Comment (flepied) [reply README.md]: ok changing that

Comment (flepied) [reply README.md]: but it works for me without it btw...

File: monocle/db/db.py (hunk 1 above)
Comment (flepied) [reply monocle/db/db.py]: ok changing that

Comment (flepied) [reply monocle/db/db.py]: done

File: README.md (hunk 4 above)
Comment (flepied) [reply README.md]: done

**Pull request merged**